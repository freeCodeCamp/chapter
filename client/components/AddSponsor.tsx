import React, { useState } from 'react';
import { useForm, Controller } from 'react-hook-form';
import {
  Button,
  TextField,
  makeStyles,
  Select,
  MenuItem,
  InputLabel,
  FormControl,
} from '@material-ui/core';

const useStyles = makeStyles(() => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '25%',
  },
  item: {
    marginTop: '20px',
  },
  responseDiv: {
    margin: '15px 0',
  },
}));

const AddSponsor: React.FC = () => {
  const [responseMsg, setResponseMsg] = useState('');
  const styles = useStyles();

  const { control, handleSubmit } = useForm();

  // TODO: Get data from store
  // const eventId = useSelector(state => state.selectedChapter.eventId);
  // const chapterId = useSelector(state => state.selectedChapter.id);

  const onSubmit = async data => {
    try {
      // await dispatch(sponsorActions.submit(eventId, chapterId));
      setResponseMsg(`${data.name} has been added as a ${data.type} sponsor.`);
    } catch (e) {
      setResponseMsg('Uh oh, something went wrong.');
      // TODO: more descriptive error messages
    }
  };

  return (
    <>
      {responseMsg && <div className={styles.responseDiv}>{responseMsg}</div>}
      <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
        <FormControl className={styles.item}>
          <Controller
            control={control}
            as={
              <TextField
                name="name"
                type="text"
                label="Sponsor Name"
                placeholder="Glitter and Sparkle Co"
              />
            }
            name="name"
            options={{ required: true }}
          />
        </FormControl>
        <FormControl className={styles.item}>
          <Controller
            control={control}
            as={
              <TextField
                label="Sponsor Website"
                name="website"
                type="text"
                placeholder="www.glitter.co"
              />
            }
            name="website"
            options={{ required: true }}
          />
        </FormControl>
        <FormControl className={styles.item}>
          <InputLabel id="sponsor-type-label">Sponsor Type</InputLabel>
          <Controller
            control={control}
            as={
              <Select labelId="sponsor-type-label">
                <MenuItem value={'FOOD'}>Food</MenuItem>
                <MenuItem value={'BEVERAGE'}>Beverage</MenuItem>
                <MenuItem value={'OTHER'}>Other</MenuItem>
              </Select>
            }
            name="type"
            defaultValue={'OTHER'}
            options={{ required: true }}
          />
        </FormControl>
        <Button
          className={styles.item}
          variant="contained"
          color="primary"
          type="submit"
        >
          Add Sponsor
        </Button>
      </form>
    </>
  );
};

export default AddSponsor;
