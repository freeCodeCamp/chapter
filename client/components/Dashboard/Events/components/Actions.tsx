import React from 'react';
import { Button } from '@material-ui/core';
import Link from 'next/link';

import { eventActions } from 'client/store/actions';
import useConfirm from 'client/hooks/useConfirm';
import useThunkDispatch from 'client/hooks/useThunkDispatch';
import { IEventModal } from 'client/store/types/events';

interface IActionsProps {
  event: IEventModal;
  onDelete?: () => void;
}

const Actions: React.FC<IActionsProps> = ({ event, onDelete }) => {
  const dispatch = useThunkDispatch();

  const [confirmCancel, clickCancel] = useConfirm(() =>
    dispatch(eventActions.cancelEvent(1, event.id)),
  );
  const [confirmRemove, clickRemove] = useConfirm(() => {
    dispatch(eventActions.removeEvent(1, event.id));
    onDelete && onDelete();
  });

  return (
    <>
      {!event.canceled && (
        <Button onClick={clickCancel}>
          {confirmCancel ? 'Are you sure?' : 'Cancel'}
        </Button>
      )}
      <Button onClick={clickRemove}>
        {confirmRemove ? 'Are you sure?' : 'delete'}
      </Button>
      <Link
        href={`/dashboard/events/[id]/edit`}
        as={`/dashboard/events/${event.id}/edit`}
      >
        <a style={{ marginLeft: '10px' }}>Edit</a>
      </Link>
    </>
  );
};

export default Actions;
