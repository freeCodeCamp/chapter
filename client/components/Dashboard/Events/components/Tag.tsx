import React from 'react';
import { makeStyles } from '@material-ui/core';

import { ITagModal } from 'client/store/types/events';

const useStyles = makeStyles({
  tag: {
    borderRadius: '8px',
    padding: '4px 12px',
    backgroundColor: '#22c98c',
    margin: '8px 2px',
    color: '#fff',
    fontWeight: 600,
  },
  tagsContainer: { display: 'flex' },
});

const Tags: React.FC<{ tags?: ITagModal[] }> = ({ tags }) => {
  const style = useStyles();

  return tags ? (
    <div className={style.tagsContainer}>
      {tags.map(tag => (
        <div className={style.tag} key={`tag-${tag.id}`}>
          {tag.name}
        </div>
      ))}
    </div>
  ) : null;
};

export default Tags;
