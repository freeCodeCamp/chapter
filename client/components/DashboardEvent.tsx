import React from 'react';
import { Card, Typography } from '@material-ui/core';

import ProgressCardContent from './ProgressCardContent';
import { IEventModal } from 'client/store/types/events';
import Link from 'next/link';

interface IDashboardEventProps {
  event: IEventModal;
  loading: boolean;
}

const DashboardEvent: React.FC<IDashboardEventProps> = ({ event, loading }) => {
  return (
    <Card style={{ marginTop: '12px' }}>
      <ProgressCardContent loading={loading}>
        <Link
          href="/dashboard/events/[id]"
          as={`/dashboard/events/${event.id}`}
        >
          <a>
            <Typography gutterBottom variant="h5" component="h2">
              {event.name}
            </Typography>
          </a>
        </Link>
        <Typography variant="body2" color="textSecondary" component="p">
          {event.description}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          {event.capacity}
        </Typography>
      </ProgressCardContent>
    </Card>
  );
};

export default DashboardEvent;
