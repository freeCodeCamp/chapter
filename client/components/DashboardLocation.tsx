import React, { useState } from 'react';
import {
  Card,
  Typography,
  Button,
  makeStyles,
  Link as MULink,
} from '@material-ui/core';
import { useDispatch } from 'react-redux';
import Link from 'next/link';

import ProgressCardContent from './ProgressCardContent';
import { ILocationModal } from 'client/store/types/locations';
import { locationActions } from 'client/store/actions';

interface IDashboardLocationProps {
  location: ILocationModal;
  loading: boolean;
}

const useStyles = makeStyles(() => ({
  bottom: {
    display: 'flex',
    alignItems: 'center',
  },
  action: {
    marginLeft: '20px',
  },
}));

const DashboardLocation: React.FC<IDashboardLocationProps> = ({
  location,
  loading,
}) => {
  const [confirm, setConfirm] = useState<boolean>(false);
  const dispatch = useDispatch();

  const styles = useStyles();

  const remove = () => {
    dispatch(locationActions.remove(location.id));
  };

  return (
    <Card style={{ marginTop: '12px' }}>
      <ProgressCardContent loading={loading}>
        <MULink gutterBottom variant="h5" component="h2">
          <Link href={`/dashboard/locations/${location.id}`}>
            {location.city}
          </Link>
        </MULink>
        <Typography variant="body2" color="textSecondary" component="p">
          {`${location.region}, ${location.country_code}, ${location.postal_code}`}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          {location.address}
        </Typography>

        <div className={styles.bottom}>
          {confirm ? (
            <Button onClick={remove}>Are you sure</Button>
          ) : (
            <Button onClick={() => setConfirm(true)}>DELETE</Button>
          )}

          <Link href={`/dashboard/locations/${location.id}/edit`}>
            <a className={styles.action}>Edit</a>
          </Link>
        </div>
      </ProgressCardContent>
    </Card>
  );
};

export default DashboardLocation;
