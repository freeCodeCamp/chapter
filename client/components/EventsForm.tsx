import React from 'react';
import { useForm, Controller } from 'react-hook-form';
import { FormControl, TextField, Button, makeStyles } from '@material-ui/core';

interface IField {
  key: string;
  label: string;
  placeholder?: string;
  type: string;
  defaultValue?: string;
}

const fields: IField[] = [
  {
    key: 'name',
    type: 'text',
    label: 'Event title',
    placeholder: 'Foo and the Bars',
  },
  {
    key: 'description',
    type: 'text',
    label: 'Description',
    placeholder: '',
  },
  {
    key: 'capacity',
    type: 'number',
    label: 'Capacity',
    placeholder: '50',
  },
  {
    key: 'tags',
    type: 'text',
    label: 'Tags (separated by a comma)',
    placeholder: 'Foo, bar',
  },
  {
    key: 'start_at',
    type: 'datetime-local',
    label: 'Start at',
    defaultValue: new Date().toISOString().slice(0, 16),
  },
  {
    key: 'end_at',
    type: 'datetime-local',
    label: 'End at',
    defaultValue: new Date(Date.now() + 1000 * 60 * 60)
      .toISOString()
      .slice(0, 16),
  },
];

const useStyles = makeStyles(() => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '25%',
  },
  item: {
    marginTop: '20px',
  },
}));

export interface IEventFormData {
  name: string;
  description: string;
  capacity: number;
  tags: string;
}

interface IEventsFormProps {
  onSubmit: (data: IEventFormData) => void;
}

const EventsForm: React.FC<IEventsFormProps> = ({ onSubmit }) => {
  const { control, handleSubmit } = useForm();
  const styles = useStyles();

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
      {fields.map(field => (
        <FormControl className={styles.item} key={field.key}>
          <Controller
            control={control}
            as={
              <TextField
                label={field.label}
                type={field.type}
                placeholder={field.placeholder}
              />
            }
            name={field.key}
            defaultValue={field.defaultValue}
            options={{ required: true }}
          />
        </FormControl>
      ))}
      <Button
        className={styles.item}
        variant="contained"
        color="primary"
        type="submit"
      >
        Add Event
      </Button>
    </form>
  );
};

export default EventsForm;
