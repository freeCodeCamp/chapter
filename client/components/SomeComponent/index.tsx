import * as React from 'react';
import Button from '@material-ui/core/Button';
import { Grid } from '@material-ui/core';
import { withTheme, makeStyles } from '@material-ui/core/styles';

// You can also easily modify material-ui components
const useStyles = makeStyles(({ palette: { background, secondary } }) => ({
  grid: {
    background: background.default,
    color: secondary.main,
  },
}));

const SomeComponent = () => {
  const styles = useStyles();

  return (
    <Grid container>
      <Grid item xs={12} className={styles.grid}>
        This is an imported component with a Button
      </Grid>
      <Button variant="outlined">CLick Here!</Button>
    </Grid>
  );
};

export default withTheme(SomeComponent);
