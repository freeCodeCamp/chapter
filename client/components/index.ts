import SomeComponent from './SomeComponent';
import AddSponsor from './AddSponsor';
import ProgressCardContent from './ProgressCardContent';

export { SomeComponent, AddSponsor, ProgressCardContent };
