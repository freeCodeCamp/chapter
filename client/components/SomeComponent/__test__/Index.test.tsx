/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import { SomeComponent } from 'client/components';
import theme from 'styles/theme';

describe('SomeComponent', () => {
  it('should render text', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <SomeComponent />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
