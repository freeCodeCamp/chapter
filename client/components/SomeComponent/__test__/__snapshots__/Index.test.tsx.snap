// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponent should render text 1`] = `
<div
  class="MuiGrid-root MuiGrid-container"
>
  <div
    class="MuiGrid-root makeStyles-grid-1 MuiGrid-item MuiGrid-grid-xs-12"
  >
    This is an imported component with a Button
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      CLick Here!
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
